pkg/cli/cvd.go

```diff
@@ -349,15 +358,43 @@ func (c *cvdCreator) createCVDFromLocalSrcs() ([]*hoapi.CVD, error) {
 	return res.CVDs, nil
 }
 
-func credentialsFactoryFromSource(source string) (CredentialsFactory, error) {
-	switch source {
+func credentialsFactoryFromSource(opts BuildAPICredentialsOpts) (CredentialsFactory, error) {
+	switch opts.Source {
 	case NoneCredentialsSource:
 		return func() string { return "" }, nil
 	case InjectedCredentialsSource:
 		return func() string { return client.InjectedCredentials }, nil
+	case FileCredentialSource:
+		return auth(opts.FilePath)
 	default:
-		return nil, fmt.Errorf("unknown credentials source: %s", source)
+		return nil, fmt.Errorf("unknown credentials source: %s", opts.Source)
+	}
+}
+
+func jwtAuth(credential []byte) (*oauth2.Token, error) {
```

Comment:
let's have only one function here for now named: `jwtAccessToken(filepath string)` rather than having `auth` and `jwtAuth`. we can do the split later when we have the implementation for OAuth if necessary, but let's avoid early refactoring.

Suggested name `jwtAccessToken` to be specific, in this context, terms like `auth` and `jwtAuth` are too generic, we are working to get an access token.